docs/USER_GUIDE.md

```diff
@@ -0,0 +1,506 @@
+# GoVPP user guide
+
+* [Binary API generator](#binary-api-generator)
+    * [Installation](#Installation)
+        * [Using the Go toolchain](#using-the--go-toolchain)
+        * [From the source](#from-the-source)
+    * [Plugins](#plugins)
+    * [Options](#options)
+* [VPP API calls](#vpp-api-calls)
+    * [Connection](#connection)
+        * [Synchronous](#synchronous-connect)
+        * [Asynchronous](#asynchronous-connect)
+    * [Sending API messages](#sending-api-messages)
+        * [Channel](#channel)
+        * [Stream client](#stream-client)
+* [The RPC service](#the-rpc-service)
+* [VPP stats](#vpp-stats)
+    * [Low-level API connection](#low-level-stats-api-connection)
+    * [Low-level API usage](#low-level-stats-api-usage)
+
+## Binary API generator
+
+The binary API generator's purpose is to create Go bindings out of the VPP `.json` binary API definitions.
+
+### Installation
+
+Requires Go 1.18 or higher ([download](https://golang.org/dl))
+
+#### Using the  Go toolchain
+
+Latest version (most recent tag):
+
+```
+go install go.fd.io/govpp/cmd/binapi-generator@latest
+```
+
+Development version (master branch):
+
+```
+go install go.fd.io/govpp/cmd/binapi-generator@master
+```
+
+#### From the source
+
+Clone the repository:
+
+```
+git clone https://github.com/FDio/govpp
+```
+
+Call the following make taget inside:
+
+```
+make install-generator
+```
+
+# Generate API bindings
+
+At first, you need VPP JSON API bindings to build go bindings from.
+Build the VPP locally, or download
+from `packagecloud.io` ([learn more](https://fd.io/docs/vpp/master/gettingstarted/installing)).
+
+JSON API files are located in `/usr/share/vpp/api/`
+
+```
+$ make generate-binapi 
+# installing binapi-generator v0.8.0-alpha-6-g2054a76-dirty
+# generating binapi
+binapi-generator -input-dir=/usr/share/vpp/api -output-dir=. -gen=rpc
+INFO[0000] resolved VPP version from installed package: 23.06-rc0~65-g2ddb2fdaa
+INFO[0000] Generating 236 files                         
+binapi-generator -input-file=/usr/share/vpp/api/core/vpe.api.json -output-dir=. -gen=http
+INFO[0000] resolved VPP version from installed package: 23.06-rc0~65-g2ddb2fdaa
+INFO[0000] Generating 3 files   
+```
+
+Generated files can be found in `/binapi` package.
```

Comment:
This should be rewritten for User Guide according to:

- getting VPP should be separate section
- no need to build the VPP to generate bindings, just run `make json-api-files` and select folder under build-root
- using `make generate-binapi` to generate binapi is for developers working on the GoVPP, but this guide should provide steps how to generate bindings in user applications - for example creating `gen.go` file with `//go:generate binapi-generator ...`